### DISCUSSION TITLE
Baresip AEC on openwrt19.07

### DISCUSSION THREAD

**Discussion by rkdg29** [baresip/baresip#2257] [General]
Hi,

I have custom board which is having openwrt 19.07 running on it. It has baresip version 0.6.2 on it. 
I have development environment to build baresip code. I am facing echo issue while making call using baresip app. 
mic captures the voice of speaker and generates echo of the same. 

I built baresip with speex_aec module, enabled speex_aec module in baresip config and tried but it didn't work. 
When I google on different forums for this I found that speex_aec is old and discarded in new version of baresip.

I have few questions

1) is baresip speex_aec module buggy and I should avoid using it? If yes which module should I try for AEC.
2) When I tried to add pulse module to use pulseaudio+webrtc in my openwrt19.07 build it failed in build baresip-mod-pulse. 
  has any one used pulseaudio+webrtc in openwrt19.07? any reference for the same will help. 
3) New webrtc_aec module of latest baresip version needs webrtc sdk on it,has any one tried this on openwrt?

Thanks & Regards,
Rkd

**Comment by alfredh** [comment]
I can recommend `webrtc_aec.so` if your board is powerful enough to run it.
